rotkehlchen/chain/ethereum/manager.py

```diff
@@ -1278,3 +1280,80 @@ def get_basic_contract_info(self, address: ChecksumEvmAddress) -> Dict[str, Any]
 
         self.contract_info_cache[address] = info
         return info
+
+    def _update_curve_decoder(self, tx_decoder: 'EVMTransactionDecoder') -> None:
+        try:
+            curve_decoder = tx_decoder.decoders['Curve']
+        except KeyError as e:
+            raise InputError(
+                'Expected to find Curve decoder but it was not loaded. '
+                'Please open an issue on github.com/rotki/rotki/issues if you saw this.',
+            ) from e
+        new_mappings = curve_decoder.reload()
+        tx_decoder.address_mappings.update(new_mappings)
+
+    def curve_protocol_cache_is_queried(
+            self,
+            tx_decoder: Optional['EVMTransactionDecoder'],
+    ) -> bool:
+        """
+        Make sure that information that needs to be queried is queried and if not query it
+        Returns true if the cache was modified or false otherwise.
+        If the tx_decoder provide is None no information for the decoders is reloaded
+
+        Updates curve pools cache.
+        1. Deletes all previous cache values
+        2. Queries information about curve pools' addresses, lp tokens and used coins
+        3. Saves queried information in the cache in globaldb
+        """
+        # Do a query in the local chache to avoid a trip to the database every time
+        last_update_ts, decoder_updated = self.protocols_information_queried_at.get(GeneralCacheType.CURVE_LP_TOKENS, (Timestamp(0), False))  # noqa: E501
+        should_query = ts_now() - last_update_ts >= ETH_PROTOCOLS_CACHE_REFRESH
+
+        if should_query is False:
+            # if we need to reload the decoder do it
+            if decoder_updated is False and tx_decoder is not None:
+                self._update_curve_decoder(tx_decoder)
+                self.protocols_information_queried_at[GeneralCacheType.CURVE_LP_TOKENS] = (Timestamp(last_update_ts), True)  # noqa: E501
```

Comment:
though as I said you should remove the in memory cache, this is another reason to not add such complexity unless needed.

You have a bug here. You are updating in memory cache without changing the DB cache.